src/auto/credentials.rs

```diff
@@ -40,12 +40,12 @@ impl Credentials {
     }
 
     #[cfg(any(unix, feature = "dox"))]
-    pub fn get_unix_user(&self) -> Result<(), glib::Error> {
+    pub fn get_unix_user(&self) -> Result<u32, glib::Error> {
```

Comment:
Just thinking about it but shouldn't we create a `UserId` type or something?

Reply:
Even if we add it, this type need be freely convertible from/to u32 anyway.
also we support "type" change only for return now.

Reply:
True, and the fact that all unix types are 32 bits integers make it worse... But at least it'd be more easy to prevent having someone freely passing a pid as a user_id (even though they totally can).

Reply:
In `std` it's also just using a `u32`, see e.g. [std::os::unix::process::CommandExt::uid()](https://doc.rust-lang.org/std/os/unix/process/trait.CommandExt.html#tymethod.uid)

I'd keep it as it is for now.

Reply:
@GuillaumeGomez ok for you?

Reply:
Well, if the std does it (badly) this way, then it's better to align ourselves with it. So yes, it's ok for me.